### PR TITLE
Implement capacity logging to file

### PR DESCRIPTION
**Pull request by allen233333**
1, Add function appendLog to save worker id and corresponding spec to JSON file. 
2, add function logWorkers to log all worker ids which are currently in used in one instance. 
3, add function logAllWorkers to log all worker ids which which are in used in one project.

### CODE REVIEW

File: vault-dev-server/src/queue/service.ts [hunk 1]
```diff
@@ -35,9 +42,56 @@ class QueueServiceByProject implements QueueServiceImplementation {
...
+    const delimiter = this.logEntriesExist ? ',\n' : '';
+    this.logEntriesExist = true;
+    fs.appendFile(this.logFilePath, data);
+  }
+
+  async logWorkers(queue: Queue, projectUuid: string){
+    const workers = await queue.getWorkers();
+    const capacity = workers.length;
+    if(capacity > 0){
+      workers.forEach(worker => {
```
Comment (settinghead) [inline vault-dev-server/src/queue/service.ts:63]: Could just use `for (const worker of workers) {...}` here.

File: vault-dev-server/src/queue/service.ts [hunk 2]
```diff
@@ -35,9 +42,56 @@ class QueueServiceByProject implements QueueServiceImplementation {
...
+        // console.log(`Queue name: ${queue.name}, Worker ID: ${worker.id}, Status: ${worker.status}`);
+        const logEntries = workers.map(worker => ({
+          projectUuid: projectUuid,
+          queueName: queue.name,
+          workerId: worker.id
+        }))
```
Comment (settinghead) [inline vault-dev-server/src/queue/service.ts:69]: Missing semicolon. In fact please use the prettier formatter in vscode before commiting.

File: vault-interface/src/generated/capacity.ts [hunk 3]
```diff
@@ -11,6 +11,12 @@ import { Empty } from "./google/protobuf/empty";
 
 export const protobufPackage = "livestack";
 
+export interface CapacityInfo {
+  projectUuid: string;
+  instanceId: any;
+  totalCapacity: any; // 根据实际类型替换 'any'，比如 number
```
Comment (settinghead) [inline vault-interface/src/generated/capacity.ts:17]: Can this typing be more specific? In general we avoid using `any` unless we run out of options.

**Review by settinghead — CHANGES_REQUESTED** [review]

File: vault-dev-server/src/queue/service.ts (hunk 2 above)
Comment (allen233333) [reply vault-dev-server/src/queue/service.ts]: fix in new commit, not make new pr yet

File: vault-interface/src/generated/capacity.ts (hunk 3 above)
Comment (allen233333) [reply vault-interface/src/generated/capacity.ts]: fix in new commit, not make new pr yet

**Pull request merged**